src/GameState/index.tsx

```diff
@@ -43,6 +46,28 @@ export default function GameState({ navigation, route, player }: any) {
   const toast = useToast();
   let WS: any = undefined;
 
+  useEffect(() => {
+    getGame();
+    initWebSocket();
+    return () => {
+      if (WS) WS.close();
+    };
+  }, []);
+
+  useEffect(() => {
+    if (!!state.animatedMarble && !!state.moveMarbleReference) {
+      moveMarble(
+        state.moveMarbleReference.gameId,
+        state.moveMarbleReference.coordinates,
+        state.moveMarbleReference.playerForAPI,
+        state.moveMarbleReference.direction,
+        state.moveMarbleReference.token,
+      );
```

Comment:
It seems that your moveMarble function should have a single parameter -> moveMarbleReference?